sycl/include/sycl/ext/oneapi/experimental/group_sort.hpp

```diff
@@ -68,6 +68,29 @@ struct is_sorter_impl<Sorter, Group, Ptr,
 template <typename Sorter, typename Group, typename ValOrPtr>
 struct is_sorter : decltype(is_sorter_impl<Sorter, Group, ValOrPtr>::test(0)) {
 };
+
+template <typename Sorter, typename Group, typename Key, typename Value>
+struct is_key_value_sorter_impl {
+  template <typename G>
+  using is_expected_return_type =
+      typename std::is_same<std::tuple<Key, Value>,
+                            decltype(std::declval<Sorter>()(
+                                std::declval<G>(), std::declval<Key>(),
+                                std::declval<Value>()))>;
```

Comment:
I think `std::invoke_result_t` would allow to simplify this a bit.

Reply:
Thanks, simplified.